pyomo/core/tests/unit/test_derivs.py

```diff
@@ -176,15 +176,15 @@ def test_abs(self):
         m = pyo.ConcreteModel()
         m.x = pyo.Var(initialize=2.0)
         e = 2 * abs(m.x)
-        with self.assertRaises(DifferentiationException):
+        with self.assertRaisesRegexp(DifferentiationException, 'Cannot perform symbolic differentiation of abs\(x\)'):
```

Comment:
The reference string needs to be declared as a raw string to avoid errors in recent Python versions.  (I will fix that shortly)